### PR TITLE
webapp: az webapp up is not handling the empty code or unknown code case correctly

### LINKED ISSUE #8752: TypeError: unsupported operand type(s) for |: 'NoneType' and 'str' when sku=None
**Describe the bug**

Exception point is below.

```python
        sku = sku | 'F1'
```

https://github.com/Azure/azure-cli/blob/master/src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py#L2187

Exception

```
unsupported operand type(s) for |: 'NoneType' and 'str'
Traceback (most recent call last):
  File "/opt/az/lib/python3.6/site-packages/knack/cli.py", line 206, in invoke
    cmd_result = self.invocation.execute(args)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 328, in execute
    raise ex
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 386, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 377, in _run_job
    cmd_copy.exception_handler(ex)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/appservice/commands.py", line 54, in _polish_bad_errors
    raise ex
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 356, in _run_job
    result = cmd_copy(params)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 171, in __call__
    return self.handler(*args, **kwargs)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/__init__.py", line 452, in default_command_handler
    return op(**command_args)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/appservice/custom.py", line 2195, in create_deploy_webapp
    sku = sku | 'F1'
TypeError: unsupported operand type(s) for |: 'NoneType' and 'str'
```

**To Reproduce**

* python webapp on azure webapp on linux
* When the requirements.txt file is missing.

```sh
az webapp up --name "name" --location "location"
```

**Expected behavior**

Unsupport NoneType and str in Python.

I suggest the following code.

```python
        sku = sku or 'F1'
```

**Environment summary**

Ubuntu Linux and Mac OS-X.
azure-cli : 2.0.60

**Additional context**

May I PR?

### PR DESCRIPTION
**Pull request by panchagnula**
Fixes #8752 
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py [hunk 1]
```diff
@@ -2188,11 +2188,12 @@ def create_deploy_webapp(cmd, name, location=None, sku=None, dryrun=False):  # p
 
     # determine the details for app to be created from src contents
     lang_details = get_lang_from_content(src_dir)
+    print(lang_details)
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py]: get rid of this?

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py [hunk 2]
```diff
@@ -1921,7 +1921,7 @@ def create_function(cmd, resource_group_name, name, storage_account, plan=None,
                     deployment_container_image_name=None, tags=None):
     # pylint: disable=too-many-statements, too-many-branches
     if deployment_source_url and deployment_local_git:
-        raise CLIError('usage error: --deployment-source-url <url> | --deployment-local-git')
+        raise CLIError('usage error: --deployment-source-url <url> --deployment-local-git')
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py]: i think we should keep `|` unless these 2 arguments should be used together.

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py (hunk 1 above)
Comment (panchagnula) [reply src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py]: oops yes

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py (hunk 2 above)
Comment (panchagnula) [reply src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py]: Sorry that was by mistake - reverted

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py (hunk 1 above)
Comment (burkeholland) [reply src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py]: How does `get_lang_from_content` work if there is no project file? Does it determine project type based on file extension?

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py [hunk 3]
```diff
@@ -2188,11 +2188,12 @@ def create_deploy_webapp(cmd, name, location=None, sku=None, dryrun=False):  # p
...
+    print(lang_details)
     # we support E2E create and deploy for selected stacks, any other stack, set defaults for os & runtime
     # and skip deployment
     if lang_details['language'] is None:
         do_deployment = False
-        sku = sku | 'F1'
+        sku = sku or 'F1'
         os_val = OS_DEFAULT
         detected_version = '-'
         runtime_version = '-'
```
Comment (burkeholland) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py]: How do we determine the runtime if there is no project file?

Comment (panchagnula) [reply src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py]: @burkeholland what do you meanby project file?

Comment (burkeholland) [reply src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py]: Like a package.json. I’m trying to determine if up requires the package.json to determine the runtime.

**Comment by burkeholland** [comment]
Hey folks - any update on these questions?

**Comment by panchagnula** [comment]
the changes are now live. For your question For us to determine if Node ->  we need package.json to exist, we try to parse the package.json to get a runtime version, if we cannot we use the recommended default Node version -> which is  8.11 currently - we will update it to 10.14 with the next release. Also note currently Node defaults to Linux.

**Comment by johnpapa** [comment]
I’m trying to follow the flow here ... does this mean that if there is no package json file that it will default to node ?

Or are you saying if there is no package json then what happens?

**Comment by panchagnula** [comment]
if requirements.txt -> Node + Linux  -> Default Version 3.7
if package.json -> Node + Linux -> Default version 8.11
if *.csproj > .NET + windows -> Default Version 4.7
if *.html -> .NET + windows -> No runtime version set (will be determined by Kudu)
if none of the above -> No runtime version set (will be determined by Kudu) + windows

**Comment by panchagnula** [comment]
Actually with the next release Node logic is updated to 
 if package.json or server.js or index.js  -> Node + Linux -> Default version 10.14

**Pull request merged**